mlflow/langchain/retriever_wrapper.py

```diff
@@ -0,0 +1,127 @@
+"""Chain for wrapping a retriever."""
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+from langchain.callbacks.manager import AsyncCallbackManagerForChainRun, CallbackManagerForChainRun
+from langchain.chains.base import Chain
+from langchain.schema import BaseRetriever, Document
+from pydantic import Extra, Field
+
+
+class RetrieverWrapper(Chain):
```

Comment:
Can we rename this to `RetrievalChain`, since its a chain that just does retrieval? 

Otherwise, while reading this, I was confused why we even have this wrapper class (it's because this is a chain, and chains are serializable)

Reply:
Makes sense. Updated.